### PR TITLE
Remove parenteses from symbols started with '$'

### PR DESCRIPTION
**Pull request by akiramenai**

### CODE REVIEW

File: llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp [hunk 1]
```diff
@@ -31,4 +31,6 @@ SyncVMMCAsmInfo::SyncVMMCAsmInfo(const Triple &TheTriple) {
 
   SupportsDebugInformation = true;
   ExceptionsType = ExceptionHandling::SyncVM;
+  AllowDollarAtStartOfIdentifier = true;
```
Comment (hedgar2017) [inline llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp:34]: What about other parts of the identifier?

File: llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp [hunk 2]
```diff
@@ -31,4 +31,6 @@ SyncVMMCAsmInfo::SyncVMMCAsmInfo(const Triple &TheTriple) {
...
+  UseParensForDollarSignNames = false;
```
Comment (hedgar2017) [inline llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp:35]:
```
Identifier = [a-zA-Z_$] [a-zA-Z_$0-9.]*
```
There is also `.`.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

**Comment by hedgar2017** [comment]
@akiramenai could you add 3 tests for `$` and 3 for `.`? (at the beginning, at the end, and in the middle)

**Comment by zksync-admin-bot2** [comment]
Benchmark results:
```
╔═╡ Size (-%) ╞═════════════════════╡ All ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═══════════════════╡ All ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

╔═╡ Size (-%) ╞═══════════════╡ Real life ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═════════════╡ Real life ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

```

File: llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp (hunk 1 above)
Comment (akiramenai) [reply llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp:34]: It only applies to $ sign. https://llvm.org/doxygen/classllvm_1_1MCAsmInfo.html#a8835db48923d3b3c5aef9a73dd332709

File: llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp (hunk 2 above)
Comment (akiramenai) [reply llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp:35]: LLVM allows "." in a function name. It only applies parentheses for ids start with '$' as you can see from the doc page I referred.

**Comment by akiramenai** [comment]
@hedgar2017 
Though non touching by the patch itself, added tests as you requested to assert.

File: llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp (hunk 1 above)
Comment (hedgar2017) [reply llvm/lib/Target/SyncVM/MCTargetDesc/SyncVMMCAsmInfo.cpp:34]: Ok, with tests for the middle and end LGTM.

**Review by hedgar2017 — APPROVED** [review]

**Review by lialan — APPROVED** [review]
LGTM.

**Comment by zksync-admin-bot2** [comment]
Benchmark results:
```
╔═╡ Size (-%) ╞═════════════════════╡ All ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═══════════════════╡ All ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

╔═╡ Size (-%) ╞═══════════════╡ Real life ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═════════════╡ Real life ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

```

**Pull request merged**